arroyo-connectors/src/filesystem.rs

```diff
@@ -135,7 +136,13 @@ impl Connector for FileSystemConnector {
         schema: Option<&ConnectionSchema>,
     ) -> anyhow::Result<crate::Connection> {
         let write_target = if let Some(path) = opts.remove("path") {
-            Destination::FolderUri { path }
+            if let BackendConfig::Local(local_config) = BackendConfig::parse_url(&path, false)? {
+                Destination::LocalFilesystem {
+                    local_directory: local_config.path,
+                }
+            } else {
+                Destination::FolderUri { path }
```

Comment:
This should probably produce an error since this means that it's an s3/gcs path (or it would be straightforward to just configure an s3 bucket from that)

Reply:
We currently support s3 URIs via the FolderUri variant. This delegates parsing to ObjectStore, so this should fall through correctly. I've opened #345 as a fuller migration to arroyo-storage.

Reply:
I see, I was confused bythe FolderUri name that seems to imply a local filesystem